tpe/plugin/src/SimulationFeatures_TEST.cc

```diff
@@ -0,0 +1,149 @@
+
+    std::cout << " -- Plugin name: " << name << std::endl;
+
+    auto engine =
+      ignition::physics::RequestEngine3d<TestFeatureList>::From(plugin);
+    EXPECT_NE(nullptr, engine);
+    sdf::Root root;
+    const sdf::Errors &errors = root.Load(_world);
+    EXPECT_EQ(0u, errors.size());
+    const sdf::World *sdfWorld = root.WorldByIndex(0);
+    auto world = engine->ConstructWorld(*sdfWorld);
+
+    worlds.insert(world);
+  }
+  return worlds;
+}
+
+class SimulationFeatures_TEST
+  : public ::testing::Test,
+    public ::testing::WithParamInterface<std::string>
+{};
+
+TEST_P(SimulationFeatures_TEST, ShapeBoundingBox)
+{
+  const std::string library = GetParam();
+  if (library.empty())
+    return;
+
+  auto worlds = LoadWorlds(library, TEST_WORLD_DIR "/falling.world");
+
+  for (const auto &world : worlds)
+  {
+    auto sphere = world->GetModel("sphere");
+    auto sphereCollision = sphere->GetLink(0)->GetShape(0);
+    auto ground = world->GetModel("box");
+    auto groundCollision = ground->GetLink(0)->GetShape(0);
+
+    // Test the bounding boxes in the local frames
+    auto sphereAABB =
+      sphereCollision->GetAxisAlignedBoundingBox(*sphereCollision);
+
+    auto groundAABB =
+      groundCollision->GetAxisAlignedBoundingBox(*groundCollision);
+
+    EXPECT_EQ(ignition::math::Vector3d(-1, -1, -1),
+              ignition::math::eigen3::convert(sphereAABB).Min());
+    EXPECT_EQ(ignition::math::Vector3d(1, 1, 1),
+              ignition::math::eigen3::convert(sphereAABB).Max());
+    EXPECT_EQ(ignition::math::Vector3d(-50, -50, -0.5),
+              ignition::math::eigen3::convert(groundAABB).Min());
+    EXPECT_EQ(ignition::math::Vector3d(50, 50, 0.5),
+              ignition::math::eigen3::convert(groundAABB).Max());
+
+    // Test the bounding boxes in the world frames
+    sphereAABB = sphereCollision->GetAxisAlignedBoundingBox();
+    groundAABB = groundCollision->GetAxisAlignedBoundingBox();
+
+    // The sphere shape has a radius of 1.0, so its bounding box will have
+    // dimensions of 1.0 x 1.0 x 1.0. When that bounding box is transformed by
+    // a 45-degree rotation, the dimensions that are orthogonal to the axis of
+    // rotation will dilate from 1.0 to sqrt(2).
+    const double d = std::sqrt(2);
+    EXPECT_EQ(ignition::math::Vector3d(-d, -1, 2.0 - d),
+              ignition::math::eigen3::convert(sphereAABB).Min());
+    EXPECT_EQ(ignition::math::Vector3d(d, 1, 2 + d),
+              ignition::math::eigen3::convert(sphereAABB).Max());
+    EXPECT_EQ(ignition::math::Vector3d(-50*d, -50*d, -1),
+              ignition::math::eigen3::convert(groundAABB).Min());
+    EXPECT_EQ(ignition::math::Vector3d(50*d, 50*d, 0),
+              ignition::math::eigen3::convert(groundAABB).Max());
+  }
+}
+
+INSTANTIATE_TEST_CASE_P(PhysicsPlugins, SimulationFeatures_TEST,
+  ::testing::ValuesIn(ignition::physics::test::g_PhysicsPluginLibraries),); // NOLINT
```

Comment:
It looks like this test is not being run for TPE, because TPE is not listed on `g_PhysicsPluginLibraries`. :sweat_smile: 

You can add it like this:

```.diff
diff --git a/test/PhysicsPluginsList.hh b/test/PhysicsPluginsList.hh
index 7d047eb..99c80f6 100644
--- a/test/PhysicsPluginsList.hh
+++ b/test/PhysicsPluginsList.hh
@@ -29,6 +29,10 @@
 #define bullet_plugin_LIB ""
 #endif
 
+#ifndef tpe_plugin_LIB
+#define tpe_plugin_LIB ""
+#endif
+
 namespace ignition
 {
   namespace physics
@@ -38,6 +42,7 @@ namespace ignition
       const std::vector<std::string> g_PhysicsPluginLibraries = {
         dartsim_plugin_LIB,
         bullet_plugin_LIB,
+        tpe_plugin_LIB
       };
     }
   }
```

But then this will make a lot of other tests run for both DART and TPE. So shouldn't we just put these test in a common place? Right now this file is mostly a copy of `dartsim/src/SimulationFeatures_TEST.cc`.